Sources/Testing/Attachments/Test.AttachableContainer.swift

```diff
@@ -0,0 +1,37 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@_spi(Experimental)
+extension Test {
+  /// A protocol describing a type that can be attached to a test report or
+  /// written to disk when a test is run and which contains another value that
+  /// it stands in for.
+  ///
+  /// To attach an attachable value to a test report or test run output, use it
+  /// to initialize a new instance of ``Test/Attachment``, then call
+  /// ``Test/Attachment/attach(sourceLocation:)``. An attachment can only be
+  /// attached once.
+  ///
+  /// A type can conform to this protocol if it represents another type that
+  /// cannot directly conform to ``Test/Attachable``, such as a non-final class
+  /// or a type declared in a third-party module.
+  public protocol AttachableContainer<AttachableValue>: Attachable, ~Copyable {
```

Comment:
Maybe "Value" would be more succinct?
```suggestion
  public protocol AttachableContainer<Value>: Attachable, ~Copyable {
```

Reply:
It's a very generic term—and if this protocol is applied to some other type that conforms to other protocols, there's risk of conflict. I'd prefer to keep it a bit wordier to avoid those conflicts.